File: src/Quicksort.java
import java.util.Scanner;

public class Quicksort {
    public static void main(String[] args) {
        Scanner prompt = new Scanner(System.in);
        System.out.println("Digite o tamanho do array:");
        int n = prompt.nextInt();
        int[] array = new int[n];

        System.out.println("Digite os elementos do array:");
        for (int i = 0; i < n; i++) {
            array[i] = prompt.nextInt();
        }
        prompt.close();

        quicksort(array, 0, n - 1);

        System.out.println("Array ordenado:");
        for (int num : array) {
            System.out.print(num + " ");
        }
    }

    public static void quicksort(int[] arr, int low, int high) {
        if (low < high) {
            int pi = partition(arr, low, high);
            quicksort(arr, low, pi - 1);
            quicksort(arr, pi + 1, high);
        }
    }

    private static int partition(int[] arr, int low, int high) {
        int pivot = arr[high];
        int i = low - 1;
        for (int j = low; j < high; j++) {
            if (arr[j] <= pivot) {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
        int temp = arr[i + 1];
        arr[i + 1] = arr[high];
        arr[high] = temp;
        return i + 1;
    }
}

File: src/Quicksort.js
const prompt = require('prompt-sync')({ sigint: true });

function quicksort(arr, low, high) {
  if (low < high) {
    const pi = partition(arr, low, high);
    quicksort(arr, low, pi - 1);
    quicksort(arr, pi + 1, high);
  }
}

function partition(arr, low, high) {
  const pivot = arr[high];
  let i = low - 1;
  for (let j = low; j < high; j++) {
    if (arr[j] <= pivot) {
      i++;
      [arr[i], arr[j]] = [arr[j], arr[i]]; // Troca de valores usando destructuring
    }
  }
  [arr[i + 1], arr[high]] = [arr[high], arr[i + 1]]; // Troca de valores usando destructuring
  return i + 1;
}

function main() {
  const n = parseInt(prompt("Digite o tamanho do array:"));
  const array = [];

  console.log("Digite os elementos do array:");
  for (let i = 0; i < n; i++) {
    array[i] = parseInt(prompt(`Elemento ${i + 1}:`));
  }

  quicksort(array, 0, n - 1);

  console.log("Array ordenado:");
  console.log(array.join(" "));
}

main();